File: OOPs/Encapsulation/encapsulation.py
class BankAccount:
    def __init__(self, balance):
        self.__balance = balance

    def setAmount(self, amount):
        if amount < 0:
            print("Amount can't be negative")
        else:
            self.__balance += amount
    
    def getAmount(self):
        return self.__balance
    
bank = BankAccount(0)
bank.setAmount(100)
print(bank.getAmount())

File: OOPs/Encapsulation/encapsulation.cpp
#include <bits/stdc++.h>
using namespace std;

class BankAccount {
    private:
        int amount;
    public:
        BankAccount() {
            this->amount = 0;
        }

        void setAmount(int amount) {
            if(amount < 0) {
                cout << "Amount cannot be nagetive" << endl;
            } else {
                this->amount += amount;
            }
        }

        int getAmount() {
            return this->amount;
        }
};

int main() {
    BankAccount bank;
    bank.setAmount(100);
    cout << bank.getAmount() << endl;
}